src/adapter/graphql/src/queries/datasets/dataset_metadata.rs

```diff
@@ -79,77 +84,188 @@ impl DatasetMetadata {
         }
     }
 
+    // TODO: Private Datasets: tests
     /// Current upstream dependencies of a dataset
-    async fn current_upstream_dependencies(&self, ctx: &Context<'_>) -> Result<Vec<Dataset>> {
-        let (dependency_graph_service, dataset_registry) = from_catalog_n!(
+    async fn current_upstream_dependencies(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Vec<DependencyDatasetResult>> {
+        let (
+            dependency_graph_service,
+            dataset_action_authorizer,
+            dataset_entry_repository,
+            account_service,
+        ) = from_catalog_n!(
             ctx,
             dyn domain::DependencyGraphService,
-            dyn domain::DatasetRegistry
+            dyn kamu_core::auth::DatasetActionAuthorizer,
+            dyn kamu_datasets::DatasetEntryRepository,
```

Comment:
Do we have to expose repository up to GraphQL level? Feels like handler should work with a service instead.

Reply:
The deps moved to the brand new use-cases